oneflow/core/job/env_global_objects_scope.cpp

```diff
@@ -72,7 +73,9 @@ void InitLogging(const CppLoggingConf& logging_conf) {
   FLAGS_logbuflevel = logging_conf.logbuflevel();
   FLAGS_stderrthreshold = 1;  // 1=WARNING
   google::InitGoogleLogging("oneflow");
-  LocalFS()->RecursivelyCreateDirIfNotExist(FLAGS_log_dir);
+  if (!FLAGS_logtostderr || IsInDebugMode()) {
```

Comment:
是，如果不写文件从INFO级别都打印到屏幕上，不然INFO就丢了，我看INFO里面包含了很多信息和内容

Reply:
INFO一般是都放在文件的，所以默认应该不打印 INFO

Reply:
现在默认不写文件了，这些INFO不打印屏幕的话就丢了，我看INFO不少信息，时间、内存等等，默认丢弃吗。

Reply:
应该是要关闭的，要不跑一个正常的脚本，就会打印出一批日志

Reply:
你可以运行一个脚本，截图出来看看。

默认情况下 import oneflow 的效果应该是静默的，不产生任何额外信息。

Reply:
import oneflow的话，两行
```
I20221206 13:36:35.483639 608669 env_global_objects_scope.cpp:154] Using rpc backend: local
I20221206 13:36:35.581424 608669 epoll_comm_network.cpp:63] CommNet:Epoll listening on 0.0.0.0:33217
```

python3 pipeline.py 执行静态图的话，不少。
```
I20221206 13:25:23.700721 3866518 env_global_objects_scope.cpp:157] Using rpc backend: local
I20221206 13:25:23.787380 3866518 epoll_comm_network.cpp:63] CommNet:Epoll listening on 0.0.0.0:34091
W20221206 13:25:23.945109 3866518 env_global_objects_scope.cpp:279] Skip init RDMA because only one process in this group!
I20221206 13:25:23.945806 3866518 version.cpp:22] OneFlow git version: v0.8.1-515-g447d6b7057-snapshot
I20221206 13:25:23.945843 3866518 cuda_device_manager_factory.cpp:63] CUDA runtime version: 11.4
I20221206 13:25:23.945858 3866518 cuda_device_manager_factory.cpp:72] cuDNN version: 8.2.4
I20221206 13:25:23.945885 3866518 cuda_device_manager_factory.cpp:85] NCCL version: 2.12.10
I20221206 13:25:28.103739 3866518 time_util.h:95] [count log]{"loc":"[GraphCompile]PipelineGraph_0 OptimizationLogicalGraph","mem_rss":"2660.000000 MB","mem_vm":"30043.000000 MB","time_cost":"0 seconds"}
I20221206 13:25:28.113955 3866518 time_util.h:95] [count log]{"loc":"[GraphCompile]PipelineGraph_0 CompleteJob","mem_rss":"2661.000000 MB","mem_vm":"30043.000000 MB","time_cost":"0 seconds"}
I20221206 13:25:28.121080 3866518 plan_util.cpp:1139]
 Graph name PipelineGraph_0 in Rank: -1, Device: -1 needs to allocate [ 0.00256 MiB ] device memory.
   In general, Chunk id: -1  memory is [ -1e-06 MiB ] with mem_block_num = 0
        Unreused memory not eager var is  [ 0.00256 MiB ] with mem_block_num = 8
        Eager Variable Tensor total memory is [ 0 MiB ] with mem_block_num = 0
I20221206 13:25:28.121325 3866518 time_util.h:95] [count log]{"loc":"[GraphCompile]PipelineGraph_0 CompilePlan","mem_rss":"2670.000000 MB","mem_vm":"31419.000000 MB","time_cost":"0 seconds"}
I20221206 13:25:28.121380 3866518 time_util.h:95] [count log]{"loc":"[GraphCompile]PipelineGraph_0 SyncPlan","mem_rss":"2670.000000 MB","mem_vm":"31419.000000 MB","time_cost":"0 seconds"}
I20221206 13:25:28.129267 3866518 time_util.h:95] [count log]{"loc":"[GraphCompile]PipelineGraph_0 InitRuntime","mem_rss":"2671.000000 MB","mem_vm":"31467.000000 MB","time_cost":"0 seconds"}
```

nn.graph.debug()接口与之前一样，保持打屏幕
debug(0):
```
I20221206 13:38:16.425453 3914207 env_global_objects_scope.cpp:157] Using rpc backend: local
I20221206 13:38:16.528640 3914207 epoll_comm_network.cpp:63] CommNet:Epoll listening on 0.0.0.0:41995
W20221206 13:38:16.665077 3914207 env_global_objects_scope.cpp:279] Skip init RDMA because only one process in this group!
I20221206 13:38:16.665843 3914207 version.cpp:22] OneFlow git version: v0.8.1-515-g447d6b7057-snapshot
I20221206 13:38:16.665881 3914207 cuda_device_manager_factory.cpp:63] CUDA runtime version: 11.4
I20221206 13:38:16.665896 3914207 cuda_device_manager_factory.cpp:72] cuDNN version: 8.2.4
I20221206 13:38:16.665908 3914207 cuda_device_manager_factory.cpp:85] NCCL version: 2.12.10
(GRAPH:PipelineGraph_0:PipelineGraph) start building graph.
I20221206 13:38:20.381016 3914207 time_util.h:95] [count log]{"loc":"[GraphCompile]PipelineGraph_0 OptimizationLogicalGraph","mem_rss":"2655.000000 MB","mem_vm":"30043.000000 MB","time_cost":"0 seconds"}
(GRAPH:PipelineGraph_0:PipelineGraph) building graph Done! Cost time: 0.68s.

(GRAPH:PipelineGraph_0:PipelineGraph) start building plan.
I20221206 13:38:20.392668 3914207 time_util.h:95] [count log]{"loc":"[GraphCompile]PipelineGraph_0 CompleteJob","mem_rss":"2655.000000 MB","mem_vm":"30043.000000 MB","time_cost":"0 seconds"}
I20221206 13:38:20.398834 3914207 plan_util.cpp:1139]
 Graph name PipelineGraph_0 in Rank: -1, Device: -1 needs to allocate [ 0.00256 MiB ] device memory.
   In general, Chunk id: -1  memory is [ -1e-06 MiB ] with mem_block_num = 0
        Unreused memory not eager var is  [ 0.00256 MiB ] with mem_block_num = 8
        Eager Variable Tensor total memory is [ 0 MiB ] with mem_block_num = 0
I20221206 13:38:20.399080 3914207 time_util.h:95] [count log]{"loc":"[GraphCompile]PipelineGraph_0 CompilePlan","mem_rss":"2664.000000 MB","mem_vm":"31419.000000 MB","time_cost":"0 seconds"}
I20221206 13:38:20.399119 3914207 time_util.h:95] [count log]{"loc":"[GraphCompile]PipelineGraph_0 SyncPlan","mem_rss":"2664.000000 MB","mem_vm":"31419.000000 MB","time_cost":"0 seconds"}
I20221206 13:38:20.405792 3914207 time_util.h:95] [count log]{"loc":"[GraphCompile]PipelineGraph_0 InitRuntime","mem_rss":"2665.000000 MB","mem_vm":"31467.000000 MB","time_cost":"0 seconds"}
(GRAPH:PipelineGraph_0:PipelineGraph) building plan Done! Cost time: 0.02s.
```

debug(1)：
```
I20221206 13:39:15.530206 3917988 env_global_objects_scope.cpp:157] Using rpc backend: local
I20221206 13:39:15.617898 3917988 epoll_comm_network.cpp:63] CommNet:Epoll listening on 0.0.0.0:35879
W20221206 13:39:15.777642 3917988 env_global_objects_scope.cpp:279] Skip init RDMA because only one process in this group!
I20221206 13:39:15.778766 3917988 version.cpp:22] OneFlow git version: v0.8.1-515-g447d6b7057-snapshot
I20221206 13:39:15.778802 3917988 cuda_device_manager_factory.cpp:63] CUDA runtime version: 11.4
I20221206 13:39:15.778846 3917988 cuda_device_manager_factory.cpp:72] cuDNN version: 8.2.4
I20221206 13:39:15.778879 3917988 cuda_device_manager_factory.cpp:85] NCCL version: 2.12.10
(GRAPH:PipelineGraph_0:PipelineGraph) start building graph.
(GRAPH:PipelineGraph_0:PipelineGraph) start building graph builders of parameters and buffers.
(GRAPH:PipelineGraph_0:PipelineGraph) end building graph builders of parameters and buffers.
(GRAPH:PipelineGraph_0:PipelineGraph) start building graph inputs.
(INPUT:_PipelineGraph_0_input.0.0_2:tensor(..., placement=oneflow.placement(type="cuda", ranks=[0]), sbp=(oneflow.sbp.broadcast,),
       size=(4, 8), dtype=oneflow.float32))
(GRAPH:PipelineGraph_0:PipelineGraph) end building graph inputs.
(GRAPH:PipelineGraph_0:PipelineGraph) start building graph modules.
(MODULE:module_pipeline:PipelineModule())
(INPUT:_module_pipeline_input.0.0_2:tensor(..., placement=oneflow.placement(type="cuda", ranks=[0]), sbp=(oneflow.sbp.broadcast,),
       is_lazy='True', size=(4, 8), dtype=oneflow.float32))
(OUTPUT:_module_pipeline_output.0.0_2:tensor(..., placement=oneflow.placement(type="cuda", ranks=[0]), sbp=(oneflow.sbp.broadcast,),
       is_lazy='True', size=(4, 8), dtype=oneflow.float32))
(GRAPH:PipelineGraph_0:PipelineGraph) end building graph modules.
(GRAPH:PipelineGraph_0:PipelineGraph) start building graph outputs.
[WARNING](OUTPUT:_PipelineGraph_0_output.0.0_2:<class 'NoneType'>)
(GRAPH:PipelineGraph_0:PipelineGraph) end building graph outputs.
(GRAPH:PipelineGraph_0:PipelineGraph) start building graph with compile passes.
I20221206 13:39:19.622151 3917988 time_util.h:95] [count log]{"loc":"[GraphCompile]PipelineGraph_0 OptimizationLogicalGraph","mem_rss":"2664.000000 MB","mem_vm":"30048.000000 MB","time_cost":"0 seconds"}
(GRAPH:PipelineGraph_0:PipelineGraph) end building graph with compile passes.
(GRAPH:PipelineGraph_0:PipelineGraph) start re-building graph outputs for optimizatioin.
(GRAPH:PipelineGraph_0:PipelineGraph) end re-building graph outputs for optimizatioin.
(GRAPH:PipelineGraph_0:PipelineGraph) building graph Done! Cost time: 0.71s.

(GRAPH:PipelineGraph_0:PipelineGraph) start building plan.
I20221206 13:39:19.632143 3917988 time_util.h:95] [count log]{"loc":"[GraphCompile]PipelineGraph_0 CompleteJob","mem_rss":"2666.000000 MB","mem_vm":"30048.000000 MB","time_cost":"0 seconds"}
I20221206 13:39:19.638738 3917988 plan_util.cpp:1139]
 Graph name PipelineGraph_0 in Rank: -1, Device: -1 needs to allocate [ 0.00256 MiB ] device memory.
   In general, Chunk id: -1  memory is [ -1e-06 MiB ] with mem_block_num = 0
        Unreused memory not eager var is  [ 0.00256 MiB ] with mem_block_num = 8
        Eager Variable Tensor total memory is [ 0 MiB ] with mem_block_num = 0
I20221206 13:39:19.638986 3917988 time_util.h:95] [count log]{"loc":"[GraphCompile]PipelineGraph_0 CompilePlan","mem_rss":"2675.000000 MB","mem_vm":"31424.000000 MB","time_cost":"0 seconds"}
I20221206 13:39:19.639027 3917988 time_util.h:95] [count log]{"loc":"[GraphCompile]PipelineGraph_0 SyncPlan","mem_rss":"2675.000000 MB","mem_vm":"31424.000000 MB","time_cost":"0 seconds"}
I20221206 13:39:19.645530 3917988 time_util.h:95] [count log]{"loc":"[GraphCompile]PipelineGraph_0 InitRuntime","mem_rss":"2675.000000 MB","mem_vm":"31472.000000 MB","time_cost":"0 seconds"}
(GRAPH:PipelineGraph_0:PipelineGraph) building plan Done! Cost time: 0.02s.
```

Reply:
如果希望屏幕尽可能少输出的话，可以选择将屏幕输出等级提升为WARN，同时将一些信息也在代码里从INFO提升到WARN。
然后其他未被提升的信息就默认丢弃。ONEFLOW_DEBUG_MODE=1时INFO写文件，需要的就打开ONEFLOW_DEBUG_MODE去文件里看。

Reply:
> 如果希望屏幕尽可能少输出的话，可以选择将屏幕输出等级提升为WARN，同时将一些信息也在代码里从INFO提升到WARN。

嗯，我比较支持这种做法。没有 error 或者 warning 的话，默认是静默的。